public/views/cart.jsx

```diff
@@ -15,53 +15,104 @@
 
 'use strict';
 
-var Layout = require('./layout.jsx');
 var React = require('react');
+var Router = require('react-router');
 
+var ps = require('pubsub-js');
+var store = require('../store');
+var $ = require('jquery');
 
 module.exports = React.createClass({
+	getInitialState: function () {
+		return store && store.cart || this.props.cart;
+	},
+	componentDidMount: function () {
+		var self = this;
+		this.cartUpdate = ps.subscribe('cartUpdate', function (msg, items) {
+			//self.handleCartUpdate(items);
```

Comment:
Also, could the cartWidget be rendered via this cart's render function as a child component? Then a setState could pass new data to cartWidget as props and causing a cartWidget re-render?